### PR TITLE
Implement controller for mkspipeline

### LINKED ISSUE #42: Add MksPipeline
- Create controller for MksPipeline resource

### PR DESCRIPTION
**Pull request by UtkarshChaurasia**
- Create mkspipeline resource
- Generate deepcopy
- Implement controller for mkspipeline
- Create CRD for mkspipeline
- Fixes #42

### CODE REVIEW

**Review by avinal — CHANGES_REQUESTED** [review]
Looks nice, please add the copyright notices too, where it is missing.

**Review by ramessesii2 — APPROVED** [review]
lgtm

File: config/mksResourceExample/mkspipeline-eg.yaml [hunk 1]
```diff
@@ -1,9 +1,76 @@
 apiVersion: mkscontroller.example.mks/v1alpha1
 kind: MksPipeline
 metadata:
-  name: test-pipeline2
+  name: test-pipeline3
```
Comment (avinal) [inline config/mksResourceExample/mkspipeline-eg.yaml]:
```suggestion
  name: test-pipeline
```

**Review by avinal — APPROVED** [review]

**Review by ramessesii2 — APPROVED** [review]
lgtm

**Pull request merged**